fgpyo/sequence.py

```diff
@@ -164,3 +146,87 @@ def levenshtein(string1: str, string2: str) -> int:
                     matrix[i + 1][j + 1],  # Substitution
                 )
     return matrix[0][0]
+
+
+def longest_homopolymer_length(bases: str, min_hp: int = 0) -> int:
+    """Calculates the length of the longest homopolymer in the input sequence.
+
+    Args:
+        bases: the bases over which to compute
+        min_hp: do not search for homopolymers of this length or shorter
+
+    Return:
+        the length of the longest homopolymer
+    """
+    i = 0
+    # NB: if we have found a homopolymer of length `min_hp`, then we do not need
+    # to examine the last `min_hp` bases since we'll never find a longer one.
+    bases_len = len(bases)
+    while i < bases_len - min_hp:
+        base = bases[i].upper()
+        j = i + 1
+        while j < bases_len and bases[j] == base:
+            j += 1
+        min_hp = max(min_hp, j - i)
+        # skip over all the bases in the current homopolymer
+        i = j
+    return min_hp
```

Comment:
I find it a little weird that with this change, the following is true:

```
longest_homopolymer_length("ACGTACGTACGT", min_hp=3) == 3
```

The problem in general with having the `min_hp` parameter is that it becomes challenging to define what the behavior should be if no homopolymer is found of at least that length.  I.e. should this return `Optional[int]` now?